lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -253,34 +278,68 @@ def _disable_if_needed(self, plugin: OpenSearchPlugin) -> bool:
         except KeyError as e:
             raise OpenSearchPluginMissingConfigError(e)
 
+    def _compute_settings(
+        self, config: OpenSearchPluginConfig
+    ) -> Tuple[Dict[str, str], Dict[str, str]]:
+        """Returns the current and the new configuration."""
+        current_settings = ClusterTopology.get_cluster_settings(
+            self._charm.opensearch,
+            include_defaults=True,
+        )
+        to_remove = dict(
+            zip(config.config_entries_to_del, [None] * len(config.config_entries_to_del))
```

Comment:
Does this mean that the deletion of a config entry is equivalent to setting it as `None`?

Reply:
I'm probably missing something, but why not simply use the keys of `to_remove` to drop them from the dict directly? why go with `key: None`?

Reply:
No reason other than me trying to put everything on a single-liner, lol.
I will delete the keys from the cluster setting

Reply:
I can, but then I need a deepcopy of current_settings, as I use it later to build the difference between old and new configuration dicts.